test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1883,6 +1883,49 @@
            (testing "concurrent-depth metric should be 0"
              (is (= 0 (concurrent-depth))))))))))
 
+(deftest block-command-processing
+  (with-test-db
+    (svc-utils/call-with-puppetdb-instance
+      (assoc (svc-utils/create-temp-config)
+        :database *db*
+        :command-processing {:concurrent-writes 1})
```

Comment:
Hmm, not sure we need the concurrent-write limit?  That's on the ingest side iirc.  And I suspect you might be able to use `with-pdb-with-no-gc` and avoid needing to explicitly specify the database/config at all, or have the thunk/wrapper i.e. the `(fn [] ...)`.

Reply:
Changed to use `with-pdb-with-no-gc` as I don't really need to make any changes to database config.